source/textInfos/__init__.py

```diff
@@ -472,17 +477,26 @@ def getTextInChunks(self, unit):
 			yield chunkInfo.text
 			unitInfo.collapse(end=True)
 
-	def getControlFieldSpeech(self, attrs, ancestorAttrs, fieldType, formatConfig=None, extraDetail=False, reason=None):
+	def getControlFieldSpeech(
+			self, attrs, ancestorAttrs, fieldType, formatConfig=None, extraDetail=False, reason=None
+	) -> SpeechSequence:
 		# Import late to avoid circular import.
 		import speech
-		return speech.getControlFieldSpeech(attrs, ancestorAttrs, fieldType, formatConfig, extraDetail, reason)
+		sequence = speech.getControlFieldSpeech(
+			attrs, ancestorAttrs, fieldType, formatConfig, extraDetail, reason
+		)
+		# remove empty strings or empty sequences which originate from speech.getControlFieldSpeech
+		return list(item for item in sequence if bool(item))
 
 	def getControlFieldBraille(self, field, ancestors, reportStart, formatConfig):
 		# Import late to avoid circular import.
 		import braille
 		return braille.getControlFieldBraille(self, field, ancestors, reportStart, formatConfig)
 
-	def getFormatFieldSpeech(self, attrs, attrsCache=None, formatConfig=None, reason=None, unit=None, extraDetail=False , initialFormat=False, separator=None):
+	def getFormatFieldSpeech(
+			self, attrs, attrsCache=None, formatConfig=None, reason=None, unit=None, extraDetail=False,
+			initialFormat=False, separator=None

```

Comment:
Param separator can be removed. Anyone disagree with this?

Reply:
I"m not entirely sure. I think this was introduced because some speech synthesizers require double spaces between strings to make sure numbers are announced correctly.